### ISSUE TITLE
Variables in included Taskfiles

### ISSUE THREAD

**Issue by BayerSe** [go-task/task#618]
- Task version: 3.9.1
- Operating System: Ubuntu 18.04

I have found a bug in included Taskfiles. If such included Taskfiles share a common variable, these are overwritten with the last definition. If that is already reported / known: sorry!

Consider the following setup:

`a.Taskfile.yml`
```yaml
version: '3'

vars:
  name: A

tasks:
  say_hi:
    cmds:
      - echo {{.name}}
```

`b.Taskfile.yml`
```yaml
version: '3'

vars:
  name: B

tasks:
  say_hi:
    cmds:
      - echo {{.name}}
```

Both work as expected:

```shell
> task -t a.Taskfile.yml say_hi
task: [say_hi] echo A
A
> task -t b.Taskfile.yml say_hi
task: [say_hi] echo B
B
```

Now let's introduce a common Taskfile:

`Taskfile.yml`
```yaml
version: '3'

includes:
    a: a.Taskfile.yml
    b: b.Taskfile.yml
```

Now the following happens:

```shell
> task a:say_hi
task: [a:say_hi] echo B
B
> task b:say_hi
task: [b:say_hi] echo B
B
```

where I would have expected an "A" in the first call.

P.S. I wanted to say thank you for Taskfile. It's an amazing tool that helps in making my workflows so much nicer!

**Comment by kerma** [comment]
I believe this is currently "as-designed". @BayerSe, your example uses "global variables" which get merged before any task execution. From my point of view it's kinda expected behavior, but I understand how it can create confusion (global variables in general are a bit dangerous).

For time being documentation on this could be improved. For the long run we could consider a more sophisticated runtime solution, but it probably won't be a very trivial fix.

**Comment by BayerSe** [comment]
Oh, thanks for the clarification, @kerma! I had in mind that "includes" basically calls the other Taskfiles and was not aware that everything is merged into one big Taskfile.
